playground/src/pages/CodingEnvironment.tsx

```diff
@@ -242,10 +134,18 @@ end`;
   const verifyProgram = async () => {
     init().then(() => {
       try {
-        if (checkInputs(inputs) && checkOutputs(outputs, proof)) {
+        if (checkInputs(inputs)[0] && checkOutputs(outputs, proof)[0]) {
           const result = verify_program(code, inputs, outputs, proof);
           setOutput(`Verification succeeded with a security level of ${result} bits. \n \n \n`);
           toast.success("Verification successful");
+        } else {
+            if (!checkInputs(inputs)[0]) { 
+              setOutput(checkInputs(inputs)[1]);
+              toast.error("Verification failed"); 
+            } else { 
+                setOutput(checkOutputs(outputs, proof)[1]);
+                toast.error("Verification failed"); 
+              }
```

Comment:
Instead of indexing like this, I think it would be clearer to define and return an object type with a boolean `isValid` and a string "error". Also`checkInputs` is being run three times in the error case. I would just run it once and get a result. Finally, you can condense things a bit with an "else if"

Something like the following:

```
const inputCheck = checkInputs(inputs);
const outputCheck = checkOutputs(outputs, proof);
if (inputCheck.isValid && outputCheck.isValid) {
          const result = verify_program(code, inputs, outputs, proof);
          setOutput(`Verification succeeded with a security level of ${result} bits. \n \n \n`);
          toast.success("Verification successful");
} else if (!inputCheck.isValid) { 
              setOutput(inputCheck.error);
              toast.error("Verification failed"); 
} else { 
                setOutput(outputCheck.error);
                toast.error("Verification failed"); 
}
```

Also some of the indentation seems kind of wonky. Shouldn't this be handled by prettier now?

Reply:
thanks that is a good suggestion

Reply:
If the success case is unnested, would it always be executed?

Reply:
I changed it and return early